### PR TITLE
DOC: Add `nd_grid` to Array creation routines

### PR DESCRIPTION
**Pull request by ghost**
Added an entry of `nd_grid` to the list of Array creation routines, and fixed some related references. Reducing 4 warnings (gh-13114).

### CODE REVIEW

File: doc/source/reference/routines.array-creation.rst [hunk 1]
```diff
@@ -85,6 +85,7 @@ Numerical ranges
    meshgrid
    mgrid
    ogrid
+   lib.index_tricks.nd_grid
```
Comment (seberg) [inline doc/source/reference/routines.array-creation.rst:88]: ndgrid is an implementation for ogrid and mgrid and I would not consider it public API.
```suggestion
```

File: numpy/lib/index_tricks.py [hunk 2]
```diff
@@ -282,7 +284,7 @@ class OGridClass(nd_grid):
 
     See Also
     --------
-    np.lib.index_tricks.nd_grid : class of `ogrid` and `mgrid` objects
+    lib.index_tricks.nd_grid : class of `ogrid` and `mgrid` objects
```
Comment (seberg) [inline numpy/lib/index_tricks.py:287]: OK, to keep it here.  Maybe it is semi-public, although I am not actually sure there is a good reason for that.  I don't think direct use adds any real functionality.

Comment (seberg) [reply numpy/lib/index_tricks.py:287]: Yeah, I think we might as well remove it, there is no reason for direct access.

**Review by rossbar — COMMENTED** [review]
I'd vote for removing references to `nd_grid` in the class docstrings rather than adding `nd_grid` to the autosummary. The fact that `mgrid` and `ogrid` derive from `nd_grid` is mostly an implementation detail and there's very little to be gained (IMO) in linking to the parent class, which I do not believe is intended for direct use.

**Comment by mattip** [comment]
> I'd vote for removing references to nd_grid in the class docstrings 

@F3eQnxN3RriK does that make sense?

**Comment by rossbar** [comment]
IIUC this was superseded by #23727 so I will close it. Please reopen if I'm mistaken!

**Pull request closed without merging**